lib/src/main/scala/spinal/lib/MasterSlave.scala

```diff
@@ -2,87 +2,199 @@ package spinal.lib
+  *
+  * {{{
+  * val braces = master(Flow(Bool))
+  *
+  * val short = master Flow (Bool)
+  *
+  * val spaceful = master port Flow(Bool)
+  *
+  * val variadic = Flow(Bool)
+  * master(variadic)
+  * }}}
+  *
+  * The "braces" syntax is short and generic, but it uses braces.
+  *
+  * The "short" syntax is short, but it is formatted with a space between the
+  * type and its parameters, and it can be used only with:
+  *
+  *   - `Flow`
+  *   - `Stream`
+  *   - `Event`
+  *
+  * The "spaceful" syntax is generic and beatiful, but more verbose.
+  *
+  * The "variadic" syntax can be used with any number of interfaces, but can
+  * be used only if the interfaces are already declared.
+  *
+  * @see [[master]] [[slave]]
+  */
+sealed trait MS {
+
+  /** Override it to define how to apply port specification on a non-null IMasterSlave */
+  protected def applyIt[T <: IMasterSlave](i: T): T
+
+  /** Declare a port without braces, long syntax
+    *
+    * {{{
+    * val input = slave port Flow(Bool)
+    * }}}
+    *
+    * See [[MS]] for other syntax.
+    */
+  def port[T <: IMasterSlave](i: T): T =
+    if (i != null) applyIt(i)
+    else i
+
+  /** Declare a port with braces
+    *
+    * {{{
+    * val input = slave(Flow(Bool))
+    * }}}
+    *
+    * or
+    *
+    * {{{
+    * val input = Flow(Bool)
+    * slave(input)
+    * }}}
+    *
+    * See [[IODirection]] for other syntaxes.
+    */
+  def apply[T <: IMasterSlave](i: T): T = port(i)
+
+  // TODO: no? def apply[T <: IMasterSlave](data: HardType[T]): T = port(data())
 
+  /** Declare existing interfaces as ports, "variadic" syntax
+    *
+    * {{{
+    * val a = Flow(Bool)
+    * val b = Stream(Bits(5 bits))
+    * master(a, b)
+    * }}}
+    */
+  def apply(is: IMasterSlave*): Unit = is.foreach(port(_))
+
+  // TODO: this is really differen than IODirection
+  // Maybe we can compare the two means and choose the best,
+  // or document why it is not the same in comments.
   object Flow extends FlowFactory {
-    override def postApply(interface : IMasterSlave) : Unit = {
+    // TODO: can't it just be replaced by port(this)
+    override def postApply(interface: IMasterSlave): Unit = {
       super.postApply(interface)
-      MS.this.apply(interface)
+      port(interface)
     }
   }
 
   object Stream extends StreamFactory {
-    override def postApply(interface : IMasterSlave) : Unit = {
+    override def postApply(interface: IMasterSlave): Unit = {
       super.postApply(interface)
-      MS.this.apply(interface)
+      port(interface)
     }
   }
 
   object event extends EventFactory {
-    override def postApply(interface : IMasterSlave) : Unit = {
+    override def postApply(interface: IMasterSlave): Unit = {
       super.postApply(interface)
-      MS.this.apply(interface)
+      port(interface)
     }
   }
 
-  def Event : Event = {
-    val ret = spinal.lib.Event
-    this.apply(ret)
-    ret
-  }
+  // TODO: why not just renaming event above as Event?
```

Comment:
I think object event is never ever used. So could be deleted likely

Reply:
Maybe just deprecate it for now, to avoid breaking changes?

Reply:
I realy think we can just delete it, as  is can currently only be used with : 
slave.event.apply
which is kind of very broken XD

Reply:
Ok wilco